ocp_resources/rhmi.py

```diff
@@ -8,3 +9,32 @@ class RHMI(NamespacedResource):
     """
 
     api_group = NamespacedResource.ApiGroup.INTEGREATLY_ORG
+
+    def wait_for_stage_status_complete(self, timeout):
+        """
+        Wait until RHMI stage status is complete.
+
+        Args:
+            timeout (int): Time in seconds to wait for stage status.
+
+        Raises:
+            TimeoutExpiredError: If stage status is not complete.
+
+        """
+        self.logger.info(f"Wait for {self.kind} {self.name} stage status to be {self.Status.COMPLETE}")
+        sample = None
+        try:
+            timeout_watcher = TimeoutWatch(timeout=timeout)
+            self.wait(timeout=timeout_watcher.remaining_time())
+
+            for sample in TimeoutSampler(
+                wait_timeout=timeout_watcher.remaining_time(),
```

Comment:
`timeout_watcher.remaining_time()` should be saved and passed to the sampler once